docs/paper/dev/api/plugin-messaging.md

```diff
@@ -0,0 +1,183 @@
+| `Message`         | Sends a message to the specified player.               | `player name`, `message`                                         | N/A                                               |
+| `GetServer`       | Returns the server the player is connected to.         | N/A                                                              | `server name`                                     |
+| `UUID`            | Returns the UUID of player.                            | N/A                                                              | `UUID`                                            |
+| `UUIDOther`       | Returns the UUID of the specified player.              | `player name`                                                    | `player name`, `UUID`                             |
+| `ServerIp`        | Returns the IP of the specified server.                | `server name`                                                    | `server name`, `IP`, `port`                       |
+| `KickPlayer`      | Kicks the specified player.                            | `player name`, `reason`                                          | N/A                                               |
+| `Forward`         | Forwards a plugin message to another server.           | `server`, `subchannel`, `size of plugin message`, `message`      | `subchannel`, `size of plugin message`, `message` |
+| `ForwardToPlayer` | Forwards a plugin message to another player.           | `player name`, `subchannel`, `size of plugin message`, `message` | `subchannel`, `size of plugin message`, `message` |
+
+##### Example: `PlayerCount`
+
+```java
+public class MyPlugin extends JavaPlugin implements PluginMessageListener {
+    
+    @Override
+    public void onEnable() {
+        this.getServer().getMessenger().registerOutgoingPluginChannel(this, "BungeeCord");
+        this.getServer().getMessenger().registerIncomingPluginChannel(this, "BungeeCord", this);
+        
+        Player player = ...;
+        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+        out.writeUTF("PlayerCount");
+        out.writeUTF("lobby");
+        player.sendPluginMessage(this, "BungeeCord", out.toByteArray());
+        // The response will be handled in onPluginMessageReceived
+    }
+
+    @Override
+    public void onPluginMessageReceived(String channel, Player player, byte[] message) {
+        if (!channel.equals("BungeeCord")) {
+            return;
+        }
+        ByteArrayDataInput in = ByteStreams.newDataInput(message);
+        String subchannel = in.readUTF();
+        if (subchannel.equals("PlayerCount")) {
+            // This is our response to the PlayerCount request
+            String server = in.readUTF();
+            int playerCount = in.readInt();
+        }
+    }
+}
+```
+
+##### Example: `Forward`
```

Comment:
Would it be useful to at least throw some bare bone "what exactly are you trying to do" text here?
It obviously can be understood by *reading* the code below, however these docs are meant for developers learning this. It hence might be useful to give them a heads up as to what these examples try to accomplish.